### DISCUSSION TITLE
Native SectionList support for useInfiniteQuery

### DISCUSSION THREAD

**Discussion by shamilovtim** [TanStack/query#1993] [General]
useInfiniteQuery uses  a `page` object key by default for each page. This key doesn't conform to what React Native expects for SectionList. That seems like an acknowledged fact since the docs give an example of a double nested`map` in order to iterate over the infiniteQuery. I think having a customizable key instead of `page` would be nice, so we aren't having to manage 2D array mutations and re-renders. In addition, some sort of native support for flatlist/sectionlist is necessary. Some apps do require the performance improvements of these components and they shouldn't be disregarded.

**Comment by TkDodo** [comment]
I don't know what SectionList from React Native expects, but I doubt that it doesn't work with react-query. I also don't quite understand this:

> useInfiniteQuery uses a page object key by default for each page.

`useInfiniteQuery` gives you data in the form of an object which has a `pages` attribute, which in turn is an Array of the data you will find on each page. Not sure how keys are involved here or what this has to do with performance or re-renders 😕. Can you maybe elaborate a bit?
